File: clients/java/client-impl/src/main/java/com/flipkart/grayskull/models/request/BatchGetSecretsRequest.java
package com.flipkart.grayskull.models.request;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;

/**
 * Request body for the {@code POST /v1/secrets/batch} endpoint.
 */
@Getter
@AllArgsConstructor(onConstructor = @__(@JsonCreator))
@JsonIgnoreProperties(ignoreUnknown = true)
public final class BatchGetSecretsRequest {

    private final List<Entry> secrets;

    @Getter
    @AllArgsConstructor(onConstructor = @__(@JsonCreator))
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static final class Entry {
        private final String projectId;
        private final String secretName;
        private final int lastKnownVersion;
    }
}

File: clients/go/client-impl/internal/models/request/batch_get_secrets_request.go
// Package request holds wire-format request bodies sent by the Grayskull Go SDK.
//
// The JSON shapes here are part of the SDK <-> server contract and must remain
// byte-identical to the corresponding Java DTOs in com.flipkart.grayskull.models.request.
package request

// BatchGetSecretsRequest is the request body for POST /v1/secrets/batch.
//
// Mirrors com.flipkart.grayskull.models.request.BatchGetSecretsRequest.
type BatchGetSecretsRequest struct {
	// Secrets is the list of secrets the client is asking the server about.
	Secrets []BatchGetSecretsRequestEntry `json:"secrets"`
}

// BatchGetSecretsRequestEntry is one row in BatchGetSecretsRequest.Secrets,
// describing a single secret and the version the client most recently observed.
//
// Mirrors com.flipkart.grayskull.models.request.BatchGetSecretsRequest.Entry.
type BatchGetSecretsRequestEntry struct {
	// ProjectID is the project that owns the secret.
	ProjectID string `json:"projectId"`

	// SecretName is the name of the secret within the project.
	SecretName string `json:"secretName"`

	// LastKnownVersion is the highest dataVersion the client has already
	// processed; the server returns a fresher value only when its current
	// version is strictly greater.
	LastKnownVersion int32 `json:"lastKnownVersion"`
}
